configs/components/openssl-fips.rb

```diff
@@ -0,0 +1,56 @@
+component "openssl-fips" do |pkg, settings, platform|
+
+  pkg.build_requires "puppet-runtime"
+
+  openssl_fips_details = JSON.parse(File.read('configs/components/openssl-fips.json'))
+  openssl_fips_version = openssl_fips_details['version']
+  openssl_fips_location = openssl_fips_details['location']
+
+  tarball_name = "#{pkg.get_name}-#{openssl_fips_version}.#{platform.name}.tar.gz"
+  pkg.url File.join(openssl_fips_location, tarball_name)
+  pkg.sha1sum File.join(openssl_fips_location, "#{tarball_name}.sha1")
+  pkg.version openssl_fips_version
+  pkg.install_only true
+
+  pkg.add_source("file://resources/patches/openssl/openssl-fips.cnf.patch")
+
+  # Overlay openssl-fips onto puppet-runtime in /opt
+  #
+  # fipsmodule.cnf needs to be generated at install time on the host, so never
+  # include it in the package.
+  #
+  # openssl.cnf also needs to be updated at install time, so don't include it in
+  # the package. Instead ship `openssl-fips.cnf` and overwrite `openssl.cnf`
+  # with that version after the `fips.so` library is installed.
+  #
+  pkg.install do
+    [
+      "gunzip -c #{tarball_name} | #{platform.tar} --skip-old-files -C / -xf -",
```

Comment:
Trivial/ignorable: I like using --long-flags as much as portability allows. I also don't _think_ these acrobatics are needed any more since all of the tars we encounter can handle gzip tarballs:

`#{platform.tar} --skip-old-files --directory=/ --extract --file=#{tarball_name}`

Reply:
Updated